util/sparse/punch_windows.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build windows
+
+package sparse
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// punchAt for Windows also marks the file as sparse before it punches a hole in it.
+func punchAt(fd *os.File, off, size int64) error {
+	err := markAsSparseFile(fd)
+	if err != nil {
+		return err
+	}
+	fileHandle := syscall.Handle(fd.Fd())
+
+	zeroData := struct {
```

Comment:
Probably link over to: https://learn.microsoft.com/en-us/windows/win32/api/winioctl/ns-winioctl-file_zero_data_information